File: iOS Examples/Objective-C/ContinuousControl/ContinuousControl/ViewController.m
//
//  ViewController.m
//  ContinuousControl
//
//  Created by Aurelius Prochazka on 6/30/14.
//  Copyright (c) 2014 Aurelius Prochazka. All rights reserved.
//

#import "ViewController.h"
#import "AKAudioOutputPlot.h"

#import "AKTools.h"
#import "ContinuousControlConductor.h"

@implementation ViewController
{
    IBOutlet AKPropertySlider *amplitudeSlider;
    IBOutlet AKPropertySlider *modulationSlider;
    IBOutlet AKPropertySlider *modIndexSlider;
    IBOutlet AKPropertyLabel *amplitudeLabel;
    IBOutlet AKPropertyLabel *modulationLabel;
    IBOutlet AKPropertyLabel *modIndexLabel;
    IBOutlet AKAudioOutputPlot *plot;
    
    ContinuousControlConductor *conductor;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    conductor = [[ContinuousControlConductor alloc] init];
    
    amplitudeSlider.property  = amplitudeLabel.property  = conductor.tweakableInstrument.amplitude;
    modulationSlider.property = modulationLabel.property = conductor.tweakableInstrument.modulation;
    modIndexSlider.property   = modIndexLabel.property   = conductor.tweakableInstrument.modIndex;
    
    [AKManager addBinding:plot];
}

- (void)viewDidDisappear:(BOOL)animated {
    [conductor stop];
}

- (IBAction)runInstrument:(id)sender {
    [conductor start];
}

- (IBAction)stopInstrument:(id)sender {
    [conductor stop];
}

@end

File: iOS Examples/Swift/ContinuousControl/ContinuousControl/ViewController.swift
//
//  ViewController.swift
//  ContinuousControl
//
//  Created by Aurelius Prochazka on 6/30/14.
//  Copyright (c) 2014 Aurelius Prochazka. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var amplitudeSlider: AKPropertySlider!
    @IBOutlet weak var modulationSlider: AKPropertySlider!
    @IBOutlet weak var modIndexSlider: AKPropertySlider!
    @IBOutlet weak var amplitudeLabel: AKPropertyLabel!
    @IBOutlet weak var modulationLabel: AKPropertyLabel!
    @IBOutlet weak var modIndexLabel: AKPropertyLabel!
    @IBOutlet weak var plot: AKAudioOutputPlot!
    
     // Instance Variables
    var conductor = ContinuousControlConductor()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        amplitudeSlider.property = conductor.tweakableInstrument.amplitude
        amplitudeLabel.property  = conductor.tweakableInstrument.amplitude
        
        modulationSlider.property = conductor.tweakableInstrument.modulation
        modulationLabel.property  = conductor.tweakableInstrument.modulation
        
        modIndexSlider.property = conductor.tweakableInstrument.modIndex
        modIndexLabel.property  = conductor.tweakableInstrument.modIndex
        
        AKManager.addBinding(plot)
        
    }
    
    override func viewDidDisappear(animated: Bool) {
        conductor.stop()
    }
    
    
    @IBAction func runInstrument(sender: UIButton) {
        conductor.start()
    }
    
    @IBAction func stopInstrument(sender: UIButton) {
        conductor.stop()
    }

}

